pkg/export/otel/traces.go

```diff
@@ -562,6 +562,16 @@ func (tr *tracesOTELReceiver) acceptSpan(span *request.Span) bool {
 func traceAttributes(span *request.Span, optionalAttrs map[attr.Name]struct{}) []attribute.KeyValue {
 	var attrs []attribute.KeyValue
 
+	// add namespace to server and peer name across namespaces
```

Comment:
Hi @esara, 

So I think we need to be more specific with how we setup the Peer and the Host, because the meaning of the server address changes depending if we are doing a client or server span.

Essentially the logic is like this:
```
	if span.IsClientSpan() {
		clientAddr = request.SpanPeer(span) + "." + span.ServiceID.Namespace
		serverAddr = request.SpanHost(span) + "." + span.OtherNamespace
	} else {
		clientAddr = request.SpanPeer(span) + "." + span.OtherNamespace
		serverAddr = request.SpanHost(span) + "." + span.ServiceID.Namespace
	}
```

I would make helpers functions for these somewhere in traces.go and change to use the values derived by those Server and Peer helpers everywhere where we do this now:
```
			request.ClientAddr(request.SpanPeer(span)),
			request.ServerAddr(request.SpanHost(span)),

```

There are few of those case statements below in the code. I know it's more complicated, but in Kafka events we never get the client (since it's forever lost on the queue) and we hijack the Othernamespace for the clientId. We probably should fix that :).

Reply:
that is a good point only to change the server and the client address respectively I added a second commit to do that - but I kept it in the method (except for kafka spans), while this can be factored out, that would actually be more code than this... also the current request.SpanPeer/SpanHost are shared with metrics, so we would need a local method which in turns calls them - maybe better organized, but a bit more complicated to read

functionally is this correct?

Reply:
Yes, this works now, thanks! Based on what I can tell, you prefer that we don't add the additional "." + namespace when the services are in the same namespace, correct? I don't have a strong opinion on that, perhaps adding the .namespace always might be more consistent, but it's more data.